src/test/java/org/jabref/logic/integrity/IntegrityCheckTest.java

```diff
@@ -39,197 +37,23 @@
-        assertCorrect(withMode(createContext(StandardField.MONTH, "#dec#"), BibDatabaseMode.BIBTEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "#bla#"), BibDatabaseMode.BIBTEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "Dec"), BibDatabaseMode.BIBTEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "December"), BibDatabaseMode.BIBTEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "Lorem"), BibDatabaseMode.BIBTEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "10"), BibDatabaseMode.BIBTEX));
-        assertCorrect(withMode(createContext(StandardField.MONTH, "1"), BibDatabaseMode.BIBLATEX));
-        assertCorrect(withMode(createContext(StandardField.MONTH, "10"), BibDatabaseMode.BIBLATEX));
-        assertCorrect(withMode(createContext(StandardField.MONTH, "#jan#"), BibDatabaseMode.BIBLATEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "jan"), BibDatabaseMode.BIBLATEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "january"), BibDatabaseMode.BIBLATEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "January"), BibDatabaseMode.BIBLATEX));
-        assertWrong(withMode(createContext(StandardField.MONTH, "Lorem"), BibDatabaseMode.BIBLATEX));
-    }
-
-    @Test
-    void testJournaltitleChecks() {
-        assertWrong(withMode(createContext(StandardField.JOURNALTITLE, "A journal"), BibDatabaseMode.BIBLATEX));
-        assertWrong(withMode(createContext(StandardField.JOURNAL, "A journal"), BibDatabaseMode.BIBTEX));
-    }
-
-    @Test
-    void testBibtexkeyChecks() {
-        final BibDatabaseContext correctContext = createContext(InternalField.KEY_FIELD, "Knuth2014");
-        correctContext.getDatabase().getEntries().get(0).setField(StandardField.AUTHOR, "Knuth");
-        correctContext.getDatabase().getEntries().get(0).setField(StandardField.YEAR, "2014");
-        assertCorrect(correctContext);
-
-        final BibDatabaseContext wrongContext = createContext(InternalField.KEY_FIELD, "Knuth2014a");
-        wrongContext.getDatabase().getEntries().get(0).setField(StandardField.AUTHOR, "Knuth");
-        wrongContext.getDatabase().getEntries().get(0).setField(StandardField.YEAR, "2014");
-        assertWrong(wrongContext);
-    }
-
-    @Test
-    void testBracketChecks() {
-        assertCorrect(createContext(StandardField.TITLE, "x"));
-        assertCorrect(createContext(StandardField.TITLE, "{x}"));
-        assertCorrect(createContext(StandardField.TITLE, "{x}x{}x{{}}"));
-        assertWrong(createContext(StandardField.TITLE, "{x}x{}}x{{}}"));
-        assertWrong(createContext(StandardField.TITLE, "}"));
-        assertWrong(createContext(StandardField.TITLE, "{"));
-    }
-
-    @Test
-    void testAuthorNameChecks() {
-        for (Field field : FieldFactory.getPersonNameFields()) {
```

Comment:
I don't find this for loop in the new code. Was it deleted completely or did I just not find it? If it was deleted, please re-add it. -- Maybe: Is the test already existing somewhere else?

Reply:
@koppor I placed these tests in the PersonNamesCheckerTest. Do you want me to add the for loops in all the added tests there?

Reply:
I just checked. Did not find `assertCorrect(withMode(createContext(field, ""), BibDatabaseMode.BIBLATEX))` there. Could you do a double check for commit and push, please?

For the for loop, please use `@MethodSource` as described at https://www.baeldung.com/parameterized-tests-junit-5#6-method.

Reply:
I just checked. Did not find `assertCorrect(withMode(createContext(field, ""), BibDatabaseMode.BIBLATEX))` there. Could you do a double check for commit and push, please?

For the for loop, please use `@MethodSource` as described at https://www.baeldung.com/parameterized-tests-junit-5#6-method.

Reply:
@koppor I am going to add as you suggested the ` @MethodSource`  but the line you mentioned is replaced by `  assertEquals(Optional.empty(), checker.checkValue(""));` because we want interdependent test classes and the methods used in `assertCorrect(withMode(createContext(field, ""), BibDatabaseMode.BIBLATEX)) `  are only in the IntegrityCheckTest.

Reply:
@koppor I will use the `@MethodSource` as you suggested but the line you mentioned have been replaced because the methods used are declared in the IntegrityCheckTest and we want all the test classes to be interdependent.

Reply:
Good step forward. Note to self: 

- Check for `FieldFactory.getPersonNameFields()` in `PersonNamesCheckerTest`.
- Check for `Stream.of(Arguments.of` in the new code (as this was linked to from https://www.baeldung.com/parameterized-tests-junit-5#6-method)

After thinking longer, the first aspect cannot be made in `PersonChecker`.

Thus, please rewrite the test **here** to provide a `Field` from `FieldFactory.getPersonNameFields()` as source: Split the asserts up and have `field` as paramter. Thus, each `assertCorrect` and each `assertWrong` will become a separate `@Test`.

(The alternative is to provide two parameters to a  `@Test` as you did in `PersonNamesCheckerTest`).